src/components/header.component.vue

```diff
@@ -0,0 +1,395 @@
+<template>
+  <div class="HeaderWrapper">
+    <div class="HeaderBox">
+      <span class="LogoWrapper" v-bind:class="isShowCategory ? 'Active' : ''">
+        <a href="/">
+          <img src="../assets/CLASS101_Logo.svg" alt="/" />
+        </a>
+      </span>
+      <div class="SearchInput" v-bind:style="SearchInputHandler">
+        <form class="SearchContainer" v-bind:style="SearchContainerHandler">
+          <input
+            type="search"
+            placeholder="찾으시는 취미가 있으신가요?"
+            maxlength="100"
+            v-on:click="
+              isShowCategory = true
+              isShowCartText = false
+            "
+          />
+          <button
+            class="SearchInput__SearchButton"
+            v-bind:class="isShowCategory ? 'Active' : ''"
+          >
+            <img src="../assets/search_icon.svg" alt="/" />
+          </button>
+        </form>
+        <button
+          class="Search__CancelButton"
+          v-if="isShowCategory"
+          v-on:click="
+            isShowCategory = false
+            isShowCartText = true
+          "
+        >
+          취소
+        </button>
+      </div>
+      <div class="CartContainer" v-if="isShowCartText">
+        <div class="CartLink__Wrapper">
+          <a href="/cart" class="CartLink">장바구니</a>
+        </div>
+      </div>
+    </div>
+    <div class="MainSearchContainer__Wrapper" v-show="isShowCategory">
```

Comment:
v-show 는 요소 표시와 관련된 중요한 디렉티브니까 클래스보다 앞쪽에 배치하는 것이 좋습니다

```suggestion
    <div v-show="isShowCategory" class="MainSearchContainer__Wrapper">
```